### ISSUE TITLE
Data formatting for crossValidate

### ISSUE THREAD

**Issue by ellispatrick** [SydneyBioX/ClassifyR#27]
measurements <- list(clinical = clin[useImages,],
                     prop = cellProp[useImages,],
                     dist = pairDist[useImages,])

measurements <- lapply(measurements, as.data.frame) 
measurements <- lapply(measurements, function(x){rownames(x) <- paste0("sample",rownames(x));x})

It would be nice if we could be a bit more flexible in all data objects in a list needing to be the same class. I'm not a fan of character numeric rownames, but arguably we could be more flexible here too...

**Comment by DarioS** [comment]
Are you talking about

```r
setMethod("crossValidate", "list", function(measurements, ...
              # Check if the list only contains one data type
              if (measurements |> sapply(class) |> unique() |> length() != 1) {
                  stop("All assays must be of the same type (e.g. data.frame, matrix)")
              } 
              ...        ...
```

? I don't know why that check is there and I think I can simply delete it. It could easily be a mix of data frames and matrices, which is fine because they define `[` access the same way and these days both have features in columns and samples in rows.

**Comment by DarioS** [comment]
Closed by ba722d29daca99e179efd14f06b41437a1a4be8c
